packages/ckeditor5-ui/src/toolbar/toolbarview.js

```diff
@@ -630,6 +630,8 @@ class DynamicGrouping {
 		// nothing else. This happens, for instance, when the toolbar is detached from DOM and
 		// some logic adds or removes its #items.
 		if ( !this.viewElement.ownerDocument.body.contains( this.viewElement ) ) {
+			this.shouldUpdateGroupingOnNextResize = true;
```

Comment:
Oh damn it... I thought it would be easier to fix it... Also, just tested it with https://github.com/ckeditor/ckeditor5/commit/fb6b0dc9931e8513e9e4ad7a8c622e2445ba9805 being reverted and... I can't reproduce the issue 🤔  Could you check it too and confirm? You always can find some edge cases ;-D

Reply:
I guess we should close this PR, because reverting https://github.com/ckeditor/ckeditor5/commit/fb6b0dc9931e8513e9e4ad7a8c622e2445ba9805 gives us the same results...

Reply:
[fb6b0dc](https://github.com/ckeditor/ckeditor5/commit/fb6b0dc9931e8513e9e4ad7a8c622e2445ba9805) was related to #6570. So before we revert anything we need to figure out what the impact of this change is gonna be.

Reply:
Regarding #6570 issue, when I reverted the https://github.com/ckeditor/ckeditor5/commit/fb6b0dc9931e8513e9e4ad7a8c622e2445ba9805 on `master` block toolbar works fine. There's no sign of the bug I encountered before 🤔 

Any chance you could confirm that when you have time?